include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //uint32_t           minIndirectCommandsBufferOffsetAlignment;
+
+    /* FragmentShadingRateEnumsPropertiesNV *//* VK_NV_fragment_shading_rate_enums */
+    // VkSampleCountFlagBits    maxFragmentShadingRateInvocationCount;
+
+    /* MeshShaderPropertiesNV *//* VK_NV_mesh_shader */
+    //uint32_t           maxDrawMeshTasksCount;
+    //uint32_t           maxTaskWorkGroupInvocations;
+    //uint32_t           maxTaskWorkGroupSize[3];
+    //uint32_t           maxTaskTotalMemorySize;
+    //uint32_t           maxTaskOutputCount;
+    //uint32_t           maxMeshWorkGroupInvocations;
+    //uint32_t           maxMeshWorkGroupSize[3];
+    //uint32_t           maxMeshTotalMemorySize;
+    //uint32_t           maxMeshOutputVertices;
+    //uint32_t           maxMeshOutputPrimitives;
+    //uint32_t           maxMeshMultiviewViewCount;
+    //uint32_t           meshOutputPerVertexGranularity;
+    //uint32_t           meshOutputPerPrimitiveGranularity;
+
+    /* VK_NV_ray_tracing *//* useless because of VK_KHR_ray_tracing_pipeline */
+
+    /* ShaderSMBuiltinsPropertiesNV *//* VK_NV_shader_sm_builtins */
+    //uint32_t           shaderSMCount;
+    //uint32_t           shaderWarpsPerSM;
+
+    /* ShadingRateImagePropertiesNV *//* VK_NV_shading_rate_image */
+    //VkExtent2D         shadingRateTexelSize;
+    //uint32_t           shadingRatePaletteSize;
+    //uint32_t           shadingRateMaxCoarseSamples;
+
+    /* FragmentDensityMapOffsetPropertiesQCOM *//* VK_QCOM_fragment_density_map_offset */
+    //VkExtent2D         fragmentDensityOffsetGranularity;
+
+    /* Nabla */
+    uint32_t maxBufferSize;
+    uint32_t maxOptimallyResidentWorkgroupInvocations = 0u; //  its 1D because multidimensional workgroups are an illusion
+    uint32_t maxResidentInvocations = 0u; //  These are maximum number of invocations you could expect to execute simultaneously on this device.
+    asset::IGLSLCompiler::E_SPIRV_VERSION spirvVersion;
+
+    // utility functions
+    // In the cases where the workgroups synchronise with each other such as work DAGs (i.e. `CScanner`),
+    // `workgroupSpinningProtection` is meant to protect against launching a dispatch so wide that
+    // a workgroup of the next cut of the DAG spins for an extended time to wait on a workgroup from a previous one.
+    inline uint32_t computeOptimalPersistentWorkgroupDispatchSize(const uint64_t elementCount, const uint32_t workgroupSize, const uint32_t workgroupSpinningProtection=1u) const
+    {
+        assert(elementCount!=0ull && "Input element count can't be 0!");
+        const uint64_t infinitelyWideDeviceWGCount = (elementCount-1ull)/(static_cast<uint64_t>(workgroupSize)*static_cast<uint64_t>(workgroupSpinningProtection))+1ull;
+        const uint32_t maxResidentWorkgroups = maxResidentInvocations/workgroupSize;
+        return static_cast<uint32_t>(core::min<uint64_t>(infinitelyWideDeviceWGCount,maxResidentWorkgroups));
+    }
+};
```

Comment:
you're missing `VkPhysicalDeviceSubpassShadingPropertiesHUAWEI` which I have no plan to expose

Reply:
also `VK_KHR_sampler_ycbcr_conversion` which I also have no plan to expose

Reply:
`VkPhysicalDeviceSubpassShadingPropertiesHUAWEI` is not in [Extension Props](https://vulkan.gpuinfo.org/listpropertiesextensions.php)

Reply:
It has a limit, and we're talking in the scope of SPhysicalDeviceLimits 
```c
typedef struct VkPhysicalDeviceSubpassShadingPropertiesHUAWEI {
    VkStructureType    sType;
    void*              pNext;
    uint32_t           maxSubpassShadingWorkgroupSizeAspectRatio;
} VkPhysicalDeviceSubpassShadingPropertiesHUAWEI;
```
I'll add it at the end